app/modules/net_info.c

```diff
@@ -0,0 +1,176 @@
+/* 
+ * http://blog.mclemon.io/esp8266-contributing-to-the-nodemcu-ecosystem
+ * https://github.com/smcl/nodemcu-firmware/blob/cc04aaf92c1c076c30ef0b0eee43b3f924137440/app/modules/test.c
+ *
+ * extracted ping function
+ * adoptet jan 2017 for commit to nodeMCU by Wolfgang Rosner
+ */
+
+
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+
+#include "c_stdlib.h"
+
+#include "lwip/ip_addr.h"
+#include "espconn.h"
+#include "lwip/dns.h" 
+#include "lwip/app/ping.h"
+#include "lwip/raw.h"
+#include "c_stdio.h"
+
+
+
+static lua_State *gL = NULL;
+static int ping_callback_ref; 
+static int ping_host_count;
+static ip_addr_t ping_host_ip;
+
+
+void ping_received(void *arg, void *data) {
+    // this would require change of the interface 
+    // struct ping_msg *pingmsg = (struct ping_msg*)arg;
+    // struct ping_option *pingopt = pingmsg->ping_opt;
+    struct ping_option *pingopt = (struct ping_option*)arg;
+    struct ping_msg *pingmsg = pingopt->parent_msg ; 
+
+    struct ping_resp *pingresp = (struct ping_resp*)data;
+
+    char ipaddrstr[16];
+    ip_addr_t source_ip;
+    
+    source_ip.addr = pingopt->ip;
+    ipaddr_ntoa_r(&source_ip, ipaddrstr, sizeof(ipaddrstr));
+
+    if (ping_callback_ref != LUA_NOREF) {
+      lua_rawgeti(gL, LUA_REGISTRYINDEX, ping_callback_ref);
```

Comment:
Hm, lwip callback context - I wasn't aware of such implications. Though I see this practice in `mqtt` as well. E.g. [`espconn_regist_recvcb(pesp_conn, mqtt_socket_received);`](https://github.com/nodemcu/nodemcu-firmware/blob/master/app/modules/mqtt.c#L546) while [`mqtt_socket_received()`](https://github.com/nodemcu/nodemcu-firmware/blob/master/app/modules/mqtt.c#L286) heavily interacts with Lua context.
Does it mean we have a general problem in the code base in case callbacks aren't fed through the task api?